src/Microsoft.Health.Fhir.Core/Features/Search/Registry/SearchParameterStatusManager.cs

```diff
@@ -103,6 +103,20 @@ public async Task UpdateSearchParameterStatusAsync(IReadOnlyCollection<string> s
                 var searchParamUri = new Uri(uri);
 
                 var paramInfo = _searchParameterDefinitionManager.GetSearchParameter(searchParamUri);
+
+                if (checkSupported)
```

Comment:
>checkSupported [](start = 20, length = 14)

Correct me if I wrong, but this is called only if we add new search parameter.
Why we can't do that validation during adding search parameter to definition manager?

This thing is status manager, feels weird to force it do validation of something unrelated to status.

Reply:
You are correct.  This check now occurs in SearchParameterOperations, and is a better fit.